### PR TITLE
Modular installer part2

### PR DESCRIPTION
**Pull request by IceReaper**
Part 1 #20440
Part 2 #20445 
Part 3 #20439

What we have here:
- Installer downloads now using specified IPackageLoader. This allows other download formats besides .zip.
- Installer can now use custom installation sources besides cds using specified ISourceResolver.
- Installer can now use custom installation procedures besides the buildin formats using specified ISourceAction.
- All these things can be defined in the mod .dll and do not need to be pushed back to OpenRA.
- And a tiny bugfix which was listing downloads as cds. While that didnt work anyway, the above changes would otherwise produce a crash.

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/Logic/Installation/InstallFromDiscLogic.cs [hunk 1]
```diff
@@ -95,10 +93,10 @@ enum Mode { Progress, Message, List }
 		static readonly string CheckInstallLog = "check-install-log";
 
 		[TranslationReference("filename")]
-		static readonly string Extracing = "extracting-filename";
+		public static readonly string Extracing = "extracting-filename";
```
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/Logic/Installation/InstallFromDiscLogic.cs]: There's a typo here that we should probably fix - but this can be its own PR.

**Review by pchote — DISMISSED** [review]

File: OpenRA.Mods.Common/Widgets/Logic/Installation/InstallFromSourceLogic.cs [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+				var gameSources = new List<string>();
+				var digitalInstalls = new List<string>();
+
+				foreach (var source in missingSources)
+				{
+					var sourceResolver = source.ObjectCreator.CreateObject<ISourceResolver>($"{source.Type.Value}SourceResolver");
+
+					var availability = sourceResolver.GetAvailability();
+
+					if (availability == Availability.GameSource && !gameSources.Contains(source.Title))
```
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/Logic/Installation/InstallFromSourceLogic.cs]: Make these `HashSet`s instead?

**Review by pchote — DISMISSED** [review]

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/58fd3816669eb7db80bfdf124fda6f521dfd9d3f...5dddcc6840546f83be7b8773689567f09d7981e5)
